examples/memory_with_compression/_memory_storage.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+"""The memory storage base class."""
+from abc import abstractmethod
+from typing import Any, Iterable, Union
+
+from agentscope.message import Msg
+
+
+class MessageStorageBase:
+    """The message storage base class."""
+
+    @abstractmethod
+    async def add(self, messages: list[Msg], **kwargs: Any) -> None:
+        """Record the messages into the message storage."""
+
+    @abstractmethod
+    async def delete(self, *args: Any, **kwargs: Any) -> None:
+        """Delete the messages from the message storage."""
+
+    @abstractmethod
+    async def clear(self, **kwargs: Any) -> None:
+        """Clear the messages from the message storage."""
+
+    @abstractmethod
+    async def get(self, **kwargs: Any) -> list[Msg]:
```

Comment:
For `get()`, consider adding an optional `recent_n` parameter to return only the most recent N messages. This can help reduce memory usage and speed up retrieval when only recent history is needed.

Reply:
fix; add recent_n